### ISSUE TITLE
Documentation

### ISSUE THREAD

**Issue by anahata0108** [keystone-engine/keystone#462]
There's very little documentation for this project. It's hard to use because I don't know what APIs are available or how things work. There's no documentation of the C APIs, there's no documentation of the Python APIs. How do I use this? Reading the source is never the right answer for this question. I shouldn't have to dig into the source code to learn how something works. Samples only tell you how to do the things the samples do, and when they're very short they don't tell you much at all. With no documentation of the APIs, I can't go much further than the samples without reading the source code, which is a *lot* of work.

**Comment by anahata0108** [comment]
In fact, all of these projects--capstone, keystone, unicorn--have next to nothing in terms of documentation. These projects aren't usable for me as a result.

**Comment by aquynh** [comment]
did you look at the docs/, either on website, or on source code? i think
those provide good starting points.
for example: https://www.keystone-engine.org/docs/tutorial.html

frankly from top of my head, i dont remember any complains about docs of
these frameworks. this issue may be the first, after many years.
the samples provided in docs, and in source code seem to be more than
enough, so almost nobody asked how to use the frameworks.

there are more than 700 public tools built on top of 3 frameworks, so i
think they are usable.

anyway, they are open source projects, and documentation is always an
issue. feel free to contribute, thanks.

**Comment by aquynh** [comment]
each framework has a lot of sample code to learn/copy, for example: https://github.com/keystone-engine/keystone/tree/master/samples

**Comment by anahata0108** [comment]
I did look at the docs directory. Here's what it contains:

* compilation instructions (not relevant because I use my OS's package manager)
* a slide deck that repeats the examples in the repository and has no other code samples
* a quick summary of how keystone relates to llvm
* a README that describes the contents of the directory (this is genuinely, seriously appreciated! I wish more projects would do this)

I have outlined above why sample code is not as useful as proper API documentation. Maybe the folks who want proper documentation see there isn't anything, get discouraged, and don't ask. I'm probably the first one who actually wants to use these projects but doesn't want to spend their free time reading source code to figure out how it all works.

But if you don't want to write API documentation / etc., that's fine, just say so.
